src/Collection/CollectionTrait.php

```diff
@@ -421,22 +421,25 @@ public function combine($keyPath, $valuePath, $groupPath = null)
      * {@inheritDoc}
      *
      */
-    public function nest($idPath, $parentPath)
+    public function nest($idPath, $parentPath, $childrenKey = null)
     {
         $parents = [];
         $idPath = $this->_propertyExtractor($idPath);
         $parentPath = $this->_propertyExtractor($parentPath);
         $isObject = true;
+        if (empty($childrenKey)) {
```

Comment:
Please compare like  `=== null` or `isset()` here. `empty()` eats `'0'` which inevitably someone will want to use.